components/engine_rocks/src/misc.rs

```diff
@@ -274,10 +274,19 @@ impl MiscExt for RocksEngine {
     }
 
     fn pause_background_work(&self) -> Result<()> {
+        // This will make manual compaction return error instead of waiting. In practice
+        // we might want to identify this case by parsing error message.
+        self.as_inner().disable_manual_compaction();
```

Comment:
So it will abort manual compaction?

Reply:
Yes.

Reply:
Then you should mention it in comment. "make manual compaction return error" is different from "abort manual compaction".

Reply:
What's the difference, in a practical sense? Compactions that are affected by a `disable_manual_compaction` happens "after" the call, otherwise they happen before the call. Whether it is aborted or not makes no difference.

Reply:
So it will not abort manual compaction, instead it prevents issuing further manual compaction, is that right?